molecule/src/molecule_json_loader.cpp

```diff
@@ -1,68 +1,280 @@
+}
+
+void MoleculeJsonLoader::parseAtoms(const rapidjson::Value& atoms, BaseMolecule& mol)
+{
+    for (SizeType i = 0; i < atoms.Size(); i++)
+    {
+        std::string label;
+        int atom_idx = 0, charge = 0, valence = 0, radical = 0, isotope = 0, elem = 0, rsite_idx = 0;
+        const Value& a = atoms[i];
+        if (a.HasMember("attachmentPoints"))
+        {
+            int val = a["attachmentPoints"].GetInt();
+            for (int att_idx = 0; (1 << att_idx) <= val; att_idx++)
+                if (val & (1 << att_idx))
+                    mol.addAttachmentPoint(att_idx + 1, i);
+        }
+
+		/*
+        if (a.HasMember("pyramid"))
+        {
+            const Value& pyr_arr = a["pyramid"];
+            int pyramid[4];
+            for (int k = 0; k < 4; k++)
+            {
+                pyramid[k] = pyr_arr[k].GetInt();
+                if (pyramid[k] == i)
+                    pyramid[k] = -1;
+            }
+            MoleculeStereocenters::moveMinimalToEnd(pyramid);
+            mol.stereocenters.add(i, MoleculeStereocenters::ATOM_ABS, 0, pyramid);
+        }*/
+
+        if (a.HasMember("type"))
+        {
+            std::string atom_type = a["type"].GetString();
+            if (atom_type == "rg-label" && a.HasMember("$refs"))
+            {
+                std::string ref = a["$refs"][0].GetString();
+                if (ref.find("rg-") == 0 && ref.erase(0, 3).size())
+                {
+                    rsite_idx = std::stoi(ref);
+                    elem = ELEM_RSITE;
+                    label = "R";
+                }
+                else
+                    throw Error("invalid refs: %s", ref.c_str());
+            }
+            else
+                throw Error("invalid atom type: %s", atom_type.c_str());
+        }
+        else
+        {
+            label = a["label"].GetString();
+            if (label == "D")
+            {
+                elem = ELEM_H;
+                isotope = 2;
+            }
+            else if (label == "T")
+            {
+                elem = ELEM_H;
+                isotope = 3;
+            }
+            else
+            {
+                elem = Element::fromString2(label.c_str());
+                if (elem == -1)
+                {
+                    elem = ELEM_PSEUDO;
+                    if (isotope != 0)
+                    {
+                        throw Error("isotope number not allowed on pseudo-atoms");
+                    }
+                }
+            }
+        }
+
+        if (a.HasMember("charge"))
+            charge = a["charge"].GetInt();
+        if (a.HasMember("explicitValence"))
+            valence = a["explicitValence"].GetInt();
+        if (a.HasMember("radical"))
+            radical = a["radical"].GetInt();
+        if (a.HasMember("isotope"))
+            isotope = a["isotope"].GetInt();
 
+        if (_pmol)
+        {
+            atom_idx = _pmol->addAtom(elem);
+            if (charge)
```

Comment:
`if (charge)` checks not for the existence, but for the value. Why it cannot get value = `0`?

Reply:
Charge = 0 means no charge. Functionality not affected.